src/tarball.py

```diff
@@ -98,25 +98,24 @@ def _push_tarball(self, config, describe):
         os.unlink(tarball_path)
         return tarball
 
-    def _create_tarball_node(self, checkout_node, status, result):
+    def _create_tarball_node(self, checkout_node, tarball):
         node = {
             'parent': checkout_node['_id'],
             'name': 'tarball',
-            'artifacts': checkout_node['artifacts'],
+            'path': checkout_node['path'] + ['tarball'],
+            'artifacts': {
+                'tarball': urllib.parse.urljoin(self._storage_url, tarball),
+            },
             'revision': checkout_node['revision'],
-            'status': status,
-            'result': result,
+            'status': 'available',
```

Comment:
I think the tarball node should be created with `running` state first. After the execution of `_push_tarball` successfully, the state of the node should be changed to `available`.

```suggestion
            'status': 'running',
```

Reply:
Yes, that's right.  Will update the code accordingly.

Eventually this probably should become a template to run in a local shell anyway, based on `python.jinja2`.

Reply:
OK I've added an extra commit as it's more than just changing the state names.